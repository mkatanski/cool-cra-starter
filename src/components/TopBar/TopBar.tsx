import { ThemedContainer } from "components/theme/ThemedContainer";
import React from "react";
import styled from "styled-components";
import { TopBarContainerTheme as DarkTheme } from "styles/themes/dark";
import { TopBarContainerTheme as LightTheme } from "styles/themes/light";
import { Container } from "./Tab";

export type TopBarProps = React.PropsWithChildren<{}>;

const TopBarWrapper = styled.nav`
  height: 60px;
  background: ${({ theme }) => theme.elements.container.background};
  color: ${({ theme }) => theme.elements.container.text.primary};
  width: 100%;
  box-shadow: 0px 11px 10px -10px rgba(0, 0, 0, 0.58);

  display: flex;
  flex-direction: row;
  flex-wrap: nowrap;
  justify-content: flex-start;
  align-content: space-between;
  align-items: flex-start;
`;

export const TopBar: React.FC<TopBarProps> & {
  Container: typeof Container;
} = ({ children }: TopBarProps) => (
  <ThemedContainer
    themes={{
      light: LightTheme,
      dark: DarkTheme,
    }}
  >
    <TopBarWrapper>{children}</TopBarWrapper>
  </ThemedContainer>
);

TopBar.Container = Container;
