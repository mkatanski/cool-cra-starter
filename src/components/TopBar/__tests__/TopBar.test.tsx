import React from "react";
import TestRenderer from "react-test-renderer";

/* Uncomment below if needed */
// import { withThemeProvider, withDesktopSize } from 'testsHelpers';
// import { render, getByTestId } from "@testing-library/react";

import { TopBar } from "../TopBar";

/* Uncomment below if You need/want to use enzyme */
// import enzyme from "enzyme";

/* Component mock example */
// jest.mock("[module]", () => ({
//   [ComponentName]: ({children, ...rest}: React.PropsWithChildren<{}>) => (
//     <div id="[ComponentName]" {...rest}>{children}</div>
//   )
// }));

describe("components", () => {
  describe("TopBar", () => {
    it("should render TopBar component", () => {
      const tree = TestRenderer.create(<TopBar />);

      expect(tree).toMatchSnapshot();
    });
  });
});
