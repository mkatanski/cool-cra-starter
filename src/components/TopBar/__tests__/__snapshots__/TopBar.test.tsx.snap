// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components TopBar should render TopBar component 1`] = `
.c0 {
  height: 60px;
  background: #03658C;
  color: #f2f2f2;
  width: 100%;
  box-shadow: 0px 11px 10px -10px rgba(0,0,0,0.58);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-content: space-between;
  -ms-flex-line-pack: space-between;
  align-content: space-between;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
}

<nav
  className="c0"
/>
`;
